### ISSUE TITLE
[Bug]: Electron propagates GDK_BACKEND=x11 to subprocesses, breaking shell.openExternal with Firefox on Wayland

### ISSUE THREAD

**Issue by andersk** [electron/electron#28436]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

12.0.9
13.0.1
14.1.0
15.1.1

### What operating system are you using?

Other Linux

### Operating System Version

NixOS 21.05

### What arch are you using?

x64

### Last Known Working Electron version

12.0.0-beta.20

### Expected Behavior

`shell.openExternal("https://electronjs.org/")` opens a browser tab at https://electronjs.org/. (My browser is Firefox on Wayland with `MOZ_ENABLE_WAYLAND=1`, and has other tabs already open.)

### Actual Behavior

Firefox shows this error and does not open a tab:

“Firefox is already running, but is not responding. To use Firefox, you must first close the existing Firefox process, restart your device, or use a different profile.”

### Testcase Gist URL

https://gist.github.com/31ec2165f4aa806536e11d7d99d1e26f

I bisected this to v12.0.0-beta.20...v12.0.0-beta.21 (9faf23509d054062def1332500c05c8ee7fb9bb7...5dbb6356effbfc0f6e0528583fae832290c14207).

The difference between v12.0.0-beta.20’s working invocation of `xdg-open` and v12.0.0-beta.21’s failing invocation is that the latter has `GDK_BACKEND=x11` in the environment. I confirmed that this is responsible for the failure with manual invocations of

```sh
xdg-open https://electronjs.org/  # works
env GDK_BACKEND=x11 xdg-open https://electronjs.org/  # fails
```

I don’t know if Electron or Chromium need to set `GDK_BACKEND=x11` for their own purposes, but it definitely should not be propagated to subprocesses such as `xdg-open`.

**Comment by codebytere** [comment]
Looks like this was added by https://chromium-review.googlesource.com/c/chromium/src/+/2586184

**Comment by deepak1556** [comment]
Since we inherit the parent process environment for the process launch operation https://github.com/electron/electron/blob/master/shell/common/platform_util_linux.cc#L116 , we should remove `GDK_BACKEND` from the launch options environment, that can affect external app behaviors.

**Comment by vially** [comment]
> we should remove GDK_BACKEND from the launch options environment

But couldn't that potentially cause problems for users who have explicitly set `GDK_BACKEND` in their environment, but which was then [overwritten by Chromium](https://chromium-review.googlesource.com/c/chromium/src/+/2586184) as mentioned above?

I think ideally Electron should only unset/revert the`GDK_BACKEND` set by chromium while still keeping the user's original `GDK_BACKEND` (if set). Maybe something similar to this code block regarding `LC_ALL` in [electron_browser_main_parts.cc](https://github.com/electron/electron/blob/7c398062dd6fc29e859b336661f611db98c6b6c6/shell/browser/electron_browser_main_parts.cc#L311-L342).

**Full disclosure:** I'm not familiar enough with Electron so maybe I'm totally off here in which case feel free to ignore this.

**Comment by deepak1556** [comment]
Yup that is correct, I did mean removing the environment set by chromium in the parent process.

**Comment by jplatte** [comment]
Just ran into this with VSCodium 1.56. Is there a workaround until this is properly fixed?

**Comment by fabiscafe** [comment]
@jplatte at least for Firefox itself there is this workaround:
https://wiki.archlinux.org/title/Firefox#Applications_on_Wayland_can_not_launch_Firefox

**Comment by jplatte** [comment]
Actually I was getting crashes, not the "not responding" error messages. But as noted above, I already put in another workaround.

**Comment by andersk** [comment]
Still broken in 21.0.1, 22.0.0-alpha.1, 23.0.0-nightly.20221004. Still waiting for #29606.

**Comment by andersk** [comment]
Still an issue in 22.0.3, 23.0.0-beta.4, and 24.0.0-nightly.20230120. Still waiting for #29606.

**Comment by andersk** [comment]
Still broken in 24.1.2, 25.0.0-alpha.4, and 26.0.0-nightly.20230421. Still waiting for #29606/#32929.

**Comment by JoanVicens** [comment]
Hello, do you have any ETA for when it's going to be fixed?

**Comment by andersk** [comment]
Still an issue in 26.3.0, 27.0.0-beta.9, and 28.0.0-nightly.20231009. Still wilting for #29606/#32929.

**Comment by andersk** [comment]
Stoppp it, bot.

**Comment by tim-rex** [comment]
Discovered this to be an issue [here](https://gitlab.freedesktop.org/libdecor/libdecor/-/issues/65)

What's interesting is that I'm explicitly running VScode as a native Wayland application but still seeing `GDK_BACKEND=x11` regardless
```
$ cat .config/electron27-flags.conf 
--enable-features=WaylandWindowDecorations
--ozone-platform-hint=auto
```

**Comment by andersk** [comment]
bump

**Comment by andersk** [comment]
bump

**Comment by andersk** [comment]
bump

**Comment by andersk** [comment]
bump

**Comment by andersk** [comment]
bump

**Comment by andersk** [comment]
bump

**Comment by andersk** [comment]
bump

**Comment by damianremington274-blip** [comment]
Bump

**Comment by andersk** [comment]
bump

**Comment by andersk** [comment]
bump
